pixl_core/src/core/_secrets.py

```diff
@@ -0,0 +1,79 @@
+#  Copyright (c) 2022 University College London Hospitals NHS Foundation Trust
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""Handles fetching of project secrets from the Azure Keyvault"""
+
+import subprocess
+
+from azure.identity import DefaultAzureCredential
+from azure.keyvault.secrets import SecretClient
+from decouple import config
+
+
+class AzureKeyVault:
+    """Handles fetching of project secrets from the Azure Keyvault"""
+
+    def __init__(self) -> None:
+        """
+        Initialise the AzureKeyVault instance
+
+        Creates an EnvironmentCredential via AzureDefaultCredential to connect with a
+        ServicePrincipal and secret configured via environment variables.
+
+        This requires the following environment variables to be set, which will be picked up by the
+        Azure SDK:
+        - AZURE_CLIENT_ID
+        - AZURE_CLIENT_SECRET
+        - AZURE_TENANT_ID
+        - AZURE_KEY_VAULT_NAME
+        """
+        self._check_envvars()
+        self.kv_name = config("AZURE_KEY_VAULT_NAME")
+        self.client = self._connect_to_keyvault()
+
+    def _connect_to_keyvault(self) -> SecretClient:
+        key_vault_uri = f"https://{self.kv_name}.vault.azure.net"
+
+        credentials = DefaultAzureCredential()
+        return SecretClient(vault_url=key_vault_uri, credential=credentials)
+
+    def fetch_secret(self, secret_name: str) -> str:
```

Comment:
We should decorate this with `@lru_cache` so we won't hit the keyvault in a long running session